voxblox_ros/src/tsdf_server.cc

```diff
@@ -302,12 +302,25 @@ void TsdfServer::publishSlices() {
   tsdf_slice_pub_.publish(pointcloud);
 }
 
+void TsdfServer::publishMap() {
+  if (this->tsdf_map_pub_.getNumSubscribers() > 0) {
+    const bool only_updated = false;
+    timing::Timer publish_map_timer("map/update");
+    voxblox_msgs::Layer layer_msg;
+    serializeLayerAsMsg<TsdfVoxel>(this->tsdf_map_->getTsdfLayer(), only_updated,
+                                   &layer_msg);
+    this->tsdf_map_pub_.publish(layer_msg);
+    publish_map_timer.Stop();
+  }
+}
+
 void TsdfServer::publishPointclouds() {
   // Combined function to publish all possible pointcloud messages -- surface
   // pointclouds, updated points, and occupied points.
   publishAllUpdatedTsdfVoxels();
   publishTsdfSurfacePoints();
   publishTsdfOccupiedNodes();
+  publishMap();
```

Comment:
So right now the TSDF map gets published on mesh updates periodically, if these are turned on.
Instead of lumping this with publishPointclouds, can we add a new service call called publish_map or publish_layers? I think this would make a bit more sense. :)

Reply:
I added a service call to "publish_map" but left the publishing-event in updateMesh() https://github.com/ethz-asl/voxblox/blob/4be2a2f051fcc93b1463e8b3e690dc41291a9949/voxblox_ros/src/tsdf_server.cc#L318
in place